src/domain/decorators/protocol.ts

```
@@ -71,36 +71,46 @@ export function $protocol(
-											id?: { sv?: string };
-											name?: string;
-											value?: { value?: unknown }
-										}
-										const nestedKey = prop.id?.sv ?? prop.name
-										const nestedValue = prop.value?.value
-										if (nestedKey && nestedValue !== undefined) {
-											nestedProps[nestedKey] = nestedValue
+					type ExtractionResult = { success: boolean; key: string; value?: unknown }
+					let extractionResult: ExtractionResult
+					// Use Effect.TS for safe property extraction
+					extractionResult = Effect.runSync(
```

ESLint

[error] 'extractionResult' is never reassigned. Use 'const' instead.

(prefer-const)